lib/gitsh/arguments/subshell.rb

```diff
@@ -1,22 +1,22 @@
 require 'gitsh/capturing_environment'
+require 'gitsh/string_runner'
 
 module Gitsh
   module Arguments
     class Subshell
       def initialize(command, options = {})
         @command = command
-        @interpreter_factory = options.fetch(:interpreter_factory)
       end
 
       def value(env)
         capturing_env = CapturingEnvironment.new(env.clone)
-        interpreter_factory.new(capturing_env).execute(command)
+        StringRunner.run(env: capturing_env, command: command)
```

Comment:
We're doing this pretty inconsistently (e.g. the previous line calls `CapturingEnvironment.new` directly). I'm not sure if it's been worthwhile, especially because we mostly inject dependencies in tests but rely on defaults in production code.